axelor-budget/src/main/java/com/axelor/apps/budget/service/BudgetToolsServiceImpl.java

```diff
@@ -237,4 +229,74 @@ public Integer getBudgetControlLevel(Budget budget) {
           : null;
     }
   }
+
+  @Override
+  public Map<String, BigDecimal> buildMapWithAmounts(
```

Comment:
I don't understand the intention with this method
Is it used anywhere else beside in `public void computeTotals(GlobalBudget globalBudget)` or is it for a future usage for another object beside GlobalBudget ? 

If not, I think we should make this method directly modify a GlobalBudget instead of going through a map.

Reply:
Hello, it is used in GlobalBudgetService and BudgetLevelService and made for both

Reply:
Okay my bad, didn't see it